### DISCUSSION TITLE
Add TypeValue Category

### DISCUSSION THREAD

**Discussion by ataumo** [0x6d69636b/windows_hardening#52] [Ideas]
I suggest to add `TypeValue` Category in CSV files like : 

```
ID,Category,Name,Method,MethodArgument,RegistryPath,RegistryItem,ClassName,Namespace,Property,DefaultValue,RecommendedValue,TypeValue,Operator,Severity
1000,Features,"SMBv1 Support",WindowsOptionalFeature,smb1protocol,,,,,,Disabled,Disabled,string,=,High
...
1101,"Account Policies","Account lockout duration",accountpolicy,,,,,,,30,15,integer,>=,Low
...
1302,"Security Options","Interactive logon: Do not require CTRL+ALT+DEL",Registry,,HKLM:\Software\Microsoft\Windows\CurrentVersion\Policies\System,DisableCAD,,,,1,0,boolean,=,Low
```

**Comment by 0x6d69636b** [comment]
What should TypeValue be used for?

**Reply by ataumo** [reply]
It's because I'm working also on a macOS hardening based on your implementation (with policies in csv file configuration) and I wanted to use the same structure of csv file to keep the same parser in the hardening interface. But it's not possible because Windows and macOS configuration are really different.
Finally, for Windows implementation, it's not necessary. 🙂

**Reply by 0x6d69636b** [reply]
I would be very surprised if this worked the same under Windows and macOS. ;-)

A discussion about the data format or the use of a database has started in [issue 50](https://github.com/0x6d69636b/windows_hardening/issues/50). This could also be relevant for you and this topic.
